app/controllers/downloads_controller.rb

```diff
@@ -2,16 +2,27 @@
 
 # DownloadsController: for handling user downloads
 class DownloadsController < ApplicationController
-  before_action :authenticate_user!
+  include Rails.application.routes.url_helpers
+  before_action :authenticate_user!, :current_user_download
 
   def download
     redirect_to root_url unless current_user.can_export?
-    lookup = params[:lookup]
-    download = current_user.downloads.find_by(lookup: lookup)
-    if download.nil?
-      @error = true
-    else
-      send_data(Base64.decode64(download.contents), filename: download.filename) && download.delete
-    end
+    @export_url = rails_storage_proxy_path(@download.exports.first)
+  end
+
+  # Hit this endpoint after the download link is clicked to remove download from database
+  # and object storage
+  def downloaded
+    # This background job cannot be run when executing system tests because the exported file will be deleted before the system test can download it.
+    return if @download.marked_for_deletion || Rails.configuration.x.executing_system_tests
+
+    DestroyDownloadsJob.set(wait_until: ADMIN_OPTIONS['download_destroy_wait_time'].to_i.hour.from_now).perform_later(@download)
```

Comment:
It might actually be worth making this minutes instead of hours. That granularity would let us change this to 15 minutes if we wanted (or use `to_f` instead of `to_i`, which means we could set this as 0.25 for 15 minutes for example).

Reply:
Changed to mins